### PR TITLE
#1157 Fixed user modifications of 'User-Agent' http header on requests

### PR DESCRIPTION
**Pull request by kkrushnyakov**
The details are here:
https://github.com/dropwizard/dropwizard/issues/1157
https://groups.google.com/forum/#!topic/dropwizard-user/XmSTpzz8Xx8

This commit is a part of UpWork.com initiative to maintain the 0.8.x
branch. The details about it are here:
https://groups.google.com/forum/#!topic/dropwizard-dev/hlK_Wi_2jis

### CODE REVIEW

File: dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+
+import org.glassfish.jersey.client.ClientRequest;
+
+/**
+ * Prevents Jersey from modification Request's User-Agent header with default value, to escape the value conflict with Dropwizard
+ * 
+ * @author kirillkrushnyakov@cloud.upwork.com
```
Comment (jplock) [inline dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java]: Please remove the `@author` line

File: dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ */
+
```
Comment (jplock) [inline dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java]: Do you need a `@Provider` annotation?

File: dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java (hunk 1 above)
Comment (kkrushnyakov) [inline dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java]: Done

File: dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java (hunk 2 above)
Comment (kkrushnyakov) [inline dropwizard-client/src/main/java/io/dropwizard/client/JerseyIgnoreRequestUserAgentHeaderFilter.java]: Done

**Comment by jplock** [comment]
Sorry one last thing, can you please use 4 spaces for indentation instead of tabs?  It looks like tabs in the PR.

**Comment by jplock** [comment]
Is there anyway to provide a test for this so its not broken in the future?

**Comment by joschi** [comment]
LGTM. Thanks for adding the test cases!

**Pull request merged**